File: fm-domain/src/main/java/io/zeitwert/fm/task/model/DocTask.java
package io.zeitwert.fm.task.model;

import java.time.OffsetDateTime;

import io.dddrive.core.ddd.model.Aggregate;
import io.dddrive.core.doc.model.Doc;
import io.zeitwert.fm.account.model.ObjAccount;
import io.zeitwert.fm.task.model.enums.CodeTaskPriority;

public interface DocTask extends Doc {

	Integer getRelatedToId();

	void setRelatedToId(Integer id);

	Aggregate getRelatedTo();

	ObjAccount getAccount();

	String getSubject();

	void setSubject(String subject);

	String getContent();

	void setContent(String content);

	Boolean getIsPrivate();

	void setIsPrivate(Boolean isPrivate);

	CodeTaskPriority getPriority();

	void setPriority(CodeTaskPriority priority);

	OffsetDateTime getDueAt();

	void setDueAt(OffsetDateTime dueAt);

	OffsetDateTime getRemindAt();

	void setRemindAt(OffsetDateTime remindAt);

}

File: fm-domain/src/main/java/io/zeitwert/fm/task/model/DocTask.kt
package io.zeitwert.fm.task.model

import io.dddrive.core.ddd.model.Aggregate
import io.dddrive.core.doc.model.Doc
import io.zeitwert.fm.account.model.ItemWithAccount
import io.zeitwert.fm.collaboration.model.ItemWithNotes
import io.zeitwert.fm.task.model.enums.CodeTaskPriority
import java.time.OffsetDateTime

interface DocTask :
	Doc,
	ItemWithAccount,
	ItemWithNotes {

	var relatedToId: Any?

	val relatedTo: Aggregate?

	var subject: String?

	var content: String?

	var isPrivate: Boolean?

	var priority: CodeTaskPriority?

	var dueAt: OffsetDateTime?

	var remindAt: OffsetDateTime?

}
